packages/squiggle-lang/src/utility/E_A.ts

```diff
@@ -119,3 +119,27 @@ export function isEqual<T>(arr1: readonly T[], arr2: readonly T[]): boolean {
 
   return true;
 }
+
+export function sample<T>(array: readonly T[], rng: PRNG): T {
+  const index = Math.floor(rng() * array.length);
+  return array[index];
+}
+
+export function sampleN<T>(array: readonly T[], n: number, rng: PRNG): T[] {
+  const size = Math.max(0, Math.floor(n));
+  if (size === 0) return [];
+  if (size >= array.length) return shuffle(array, rng);
```

Comment:
Sampling 10 items from the list with 3 items will return the list of 3 items, is this intentional?

I'd expect it to create duplicates, and that would be consistent with [how Dist.sampleN works](https://www.squiggle-language.com/playground?v=dev#code=eNqrVkpJTUsszSlxzk9JVbJSCk7MLchJDU4t0Usrys%2F1ySwu0Yg21DHSMdYx0THVMYvVVNC1U3ABCusVg1X6aRgaaCrVAgBN0BYw)

Reply:
Ok, I see that this defaults to sampling without replacement on lower N values, so it makes sense to keep it consistent.

It's a pity that it's inconsistent with how `Dist.sampleN` works, though. I assume it was intentional, and also the reason why you didn't make un-namespaced `sampleN` non-polymorphic?

Seems like a potential footgun...

How about this: we rename the version with replacement to `pickN`, and make `sampleN` consistent for lists and dists(with replacement)?

Reply:
Good points. Agreed it's gnarly. 
Another option is to add configs. Maybe something like,
`sampleN(samples, {withReplacement: boolean, shuffle: boolean})`

Reply:
Yeah, options could work, but if we want to combine this with polymorphism on dists vs lists (which seems natural and good), and have different replacement defaults, then it's still problematic.